### PR TITLE
Fix GetSystemTextJsonSettings for .NET 6

### PR DESCRIPTION
**Pull request by Sinhk**
For some reason the reflection method for getting JsonOptions stopped working in .NET 6 (at least in my project). And I can't figure out why.

But since NSwag.Generation.AspNetCore now targets netcoreapp3.1 (and newer), it's no longer necessary to use reflection for those targets.

### CODE REVIEW

File: src/NSwag.Generation.AspNetCore/AspNetCoreOpenApiDocumentGenerator.cs [hunk 1]
```diff
@@ -80,7 +80,7 @@ dynamic GetJsonOptionsWithReflection(IServiceProvider sp)
                 }
             }
 
-#if NET5_0 || NETCOREAPP3_1
+#if NETCOREAPP3_1_OR_GREATER
```
Comment (RicoSuter) [inline src/NSwag.Generation.AspNetCore/AspNetCoreOpenApiDocumentGenerator.cs:83]: Is NETCOREAPP3_1_OR_GREATER really defined?

Comment (Sinhk) [reply src/NSwag.Generation.AspNetCore/AspNetCoreOpenApiDocumentGenerator.cs]: Apparently it's Only

**Comment by RicoSuter** [comment]
Isnt this now working with v13.14.4?

**Comment by Sinhk** [comment]
It is. 
If you don't want the changes to NSwag.Generation.AspNetCore.Tests, or want me to make a new PR for those, I guess this can be closed.

File: src/NSwag.Generation.AspNetCore/AspNetCoreOpenApiDocumentGenerator.cs [hunk 2]
```diff
@@ -156,13 +156,27 @@ public static JsonSerializerSettings GetSystemTextJsonSettings(IServiceProvider
 
             if (serviceProvider != null)
             {
+#if NETCOREAPP
```
Comment (RicoSuter) [inline src/NSwag.Generation.AspNetCore/AspNetCoreOpenApiDocumentGenerator.cs:159]: If not needed id revert the changes here in this file but keep the tests ok?

Comment (Sinhk) [reply src/NSwag.Generation.AspNetCore/AspNetCoreOpenApiDocumentGenerator.cs:159]: sure

**Comment by RicoSuter** [comment]
Thanks for the PR.
